### DISCUSSION TITLE
Sunbird RC Setup Failed in Local

### DISCUSSION THREAD

**Discussion by priyanka-TL** [Sunbird-RC/community#236] [Issues] [answered]
Like Course Certificate generation, In ML services, We have to generate improvement project certificates. You can refer to the JIRA ticket here for the improvement project certificate story 
https://project-sunbird.atlassian.net/browse/SB-28231

So we are trying to understand how it works and generate the certificate. I have followed the below guide to set up the RC
https://project-sunbird.atlassian.net/wiki/spaces/UM/pages/3085500423/Sunbird+RC+Setup

But while building the registry I got an error 
![Screenshot from 2022-06-23 10-48-03](https://user-images.githubusercontent.com/71248340/175220967-2cee8897-b6c5-492a-a106-ae043f26395f.png)

Can you please help to fix this error? I am trying to see how we can use the Sunbird RC to generate the certificate and understand how it will work for the project use case

cc - @aks30 @vijiurs @kiranharidas187 @varsha-sl @prateek-slokam @aishwaryashikshalokam

**Comment by pramodkvarma** [comment]
@coolbung @tejash-jl @dileepbapat any of you can help?

**Comment by gamemaker1** [comment]
Could you please try running `./mnvw clean install -DskipTests=true` instead?

```diff
-./mnv clean install -DskipTests=true
+./mnvw clean install -DskipTests=true
```

**Reply by priyanka-TL** [reply]
Tried getting the below error
bash: ./mnvw: No such file or directory

**Reply by gamemaker1** [reply]
Oops sorry I made a typo - it is `./mvnw`

**Reply by priyanka-TL** [reply]
Got Build failure message

![Screenshot from 2022-06-23 15-59-59](https://user-images.githubusercontent.com/71248340/175278968-16e8c91a-3b49-430a-98e8-c4f0907bcff8.png)

**Reply by gamemaker1** [reply]
Running `./mvnw clean install -DskipTests` gave me a `BUILD SUCCESS` message:

![Screenshot_2022-06-23-16-25-59_1920x1080](https://user-images.githubusercontent.com/34235681/175283319-2c005674-7ba1-436c-9f3e-f6fe18fe3a76.png)

**Reply by gamemaker1** [reply]
What command did you run?

**Reply by priyanka-TL** [reply]
Build was successfull. while running the command 'docker-compose up' I'm getting the below error
![Screenshot from 2022-06-24 09-29-49](https://user-images.githubusercontent.com/71248340/175459796-59a66d35-b71e-492c-84a5-1c0b1350d8a1.png)

**Reply by priyanka-TL** [reply]
@gamemaker1 Is there any API to list all the entities in sunbird RC?

**Reply by gamemaker1** [reply]
I think you can do it by using the `search` API (a POST request to `/api/v1/search`) but not passing any query - https://docs.sunbirdrc.dev/learn/technical-specification-draft#directory

**Reply by gamemaker1** [reply]
(I have not tested this myself)

**Reply by priyanka-TL** [reply]
@gamemaker1 can you share the full URL of search API

**Reply by gamemaker1** [reply]
`http://localhost:8081/api/v1/search` is the full URL.

**Reply by tejash-jl** [reply]
POST: `http://localhost:8081/api/v1/<entity-name>/search` 
BODY:  
```json
{"filters": {}}
```
@priyanka-TL search api

**Reply by priyanka-TL** [reply]
@tejash-jl im getting error on search api also
`curl --location --request POST 'http://localhost:8081/api/v1/TrainingCertificate/search' \
--header 'Content-Type: application/json' \
--data-raw '{
"filters": {}
}'`

Error 
`registry_1            | 04:11:52.199 [http-nio-8081-exec-9] ERROR d.s.r.c.RegistryEntityController:157 - Exception in controller while searching entities !
registry_1            | java.lang.NullPointerException: null
registry_1            | 	at dev.sunbirdrc.registry.controller.RegistryEntityController.searchEntity(RegistryEntityController.java:145)
registry_1            | 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
registry_1            | 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
registry_1            | 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
registry_1            | 	at java.lang.reflect.Method.invoke(Method.java:498)
registry_1            | 	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
registry_1            | 	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
registry_1            | 	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
registry_1            | 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
registry_1            | 	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
registry_1            | 	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
registry_1            | 	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
`

**Reply by tejash-jl** [reply]
@priyanka-TL 
Can you join the [Discord community channel](https://discord.gg/Q5mvw2mGC8) and ping me there, we can connect and resolve this.

**Comment by tejash-jl** [comment]
@priyanka-TL you can follow this sample usecase we have built to generate certificates.
https://github.com/Sunbird-RC/demo-certificate-issuance
Docker Compose: https://github.com/Sunbird-RC/demo-certificate-issuance/blob/main/docker-compose.yml
Notebook / API Documentation: https://github.com/Sunbird-RC/demo-certificate-issuance/blob/main/certificate-api.ipynb

**Reply by priyanka-TL** [reply]
@tejash-jl I have tried to generate the certificate using the existing entity TrainingCertificate
```
curl --location --request POST 'http://localhost:8081/api/v1/TrainingCertificate' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name":"Sunbird Learner", 
    "contact":"12345",
    "trainingTitle":"Sunbird RC Certificate Module",
    "note":"Ceritifcate issued"
}'
```
But I got the below error. Can you help me to fix this error 
![Screenshot from 2022-06-27 16-37-34](https://user-images.githubusercontent.com/71248340/175927834-cf6f6516-1d5b-4ba2-8369-07a4fac176d8.png)

**Comment by priyanka-TL** [comment, marked as answer]
@tejash-jl Thank you. Issue got resolved after adding the path instead of pwd in the docker-compose file
